src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -0,0 +1,358 @@
+    /// in that case.
+    /// </remarks>
+    [Export(typeof(IExperiment))]
+    internal sealed class KeybindingResetDetector : ForegroundThreadAffinitizedObject, IExperiment, IOleCommandTarget, IDisposable
+    {
+        // Flight info
+        private const string InternalFlightName = "keybindgoldbarint";
+        private const string ExternalFlightName = "keybindgoldbarext";
+        private const string KeybindingsFwLink = "https://go.microsoft.com/fwlink/?linkid=864209";
+
+        // Resharper commands and package
+        private const uint ResumeId = 707;
+        private const uint SuspendId = 708;
+        private const uint ToggleSuspendId = 709;
+        private static readonly Guid ReSharperPackageGuid = new Guid("0C6E6407-13FC-4878-869A-C8B4016C57FE");
+        private static readonly Guid ReSharperCommandGroup = new Guid("{47F03277-5055-4922-899C-0F7F30D26BF1}");
+
+        private readonly VisualStudioWorkspace _workspace;
+        private readonly SVsServiceProvider _serviceProvider;
+
+        private IExperimentationService _experimentationService;
+        private IVsUIShell _uiShell;
+        private IOleCommandTarget _oleCommandTarget;
+        private OleComponent _oleComponent;
+
+        private bool _disposedValue = false;
+        private uint _priorityCommandTargetCookie = VSConstants.VSCOOKIE_NIL;
+
+        private bool _infoBarOpen = false;
+
+        [ImportingConstructor]
+        public KeybindingResetDetector(VisualStudioWorkspace workspace, SVsServiceProvider serviceProvider)
+        {
+            _workspace = workspace;
+            _serviceProvider = serviceProvider;
+        }
+
+        public Task Initialize()
+        {
+            // Immediately bail if the user has asked to never see this bar again.
+            if (_workspace.Options.GetOption(KeybindingResetOptions.NeverShowAgain))
+            {
+                return Task.CompletedTask;
+            }
+
+            return InvokeBelowInputPriority(InitializeCore);
+        }
+
+        private void InitializeCore()
+        {
+            AssertIsForeground();
+
+            // Ensure one of the flights is enabled, otherwise bail
+            _experimentationService = _workspace.Services.GetRequiredService<IExperimentationService>();
+            if (!_experimentationService.IsExperimentEnabled(ExternalFlightName) && !_experimentationService.IsExperimentEnabled(InternalFlightName))
+            {
+                return;
+            }
+
+            var vsShell = _serviceProvider.GetService(typeof(SVsShell)) as IVsShell;
+            var hr = vsShell.IsPackageInstalled(ReSharperPackageGuid, out int extensionEnabled);
+            if (ErrorHandler.Failed(hr))
+            {
+                FatalError.ReportWithoutCrash(Marshal.GetExceptionForHR(hr));
+                return;
+            }
+
+            var currentStatus = _workspace.Options.GetOption(KeybindingResetOptions.ReSharperStatus);
+
+            if (extensionEnabled == 0)
```

Comment:
so we have 3 * 3 different cases. it would be nice if we have matrix shows of that 9 cases, which we should reset and which case not somewhere in the comment.

Reply:
or code might be simpler, if you just define bool[,] matrix[3,3] map that contains true or false for reset. and you get
var reset = matrix[lastResharperStatus, currentResharperStatus]

and centralize Option changes and showGoldbar in 1 place rather than spreading those code